contracts/truefi2/AaveBaseRateOracle.sol

```diff
@@ -0,0 +1,161 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.6.10;
+
+import {SafeMath} from "@openzeppelin/contracts/math/SafeMath.sol";
+import {IAaveLendingPool} from "./interface/IAave.sol";
+
+contract AaveBaseRateOracle {
+    using SafeMath for uint256;
+
+    uint16 public constant MAX_BUFFER_SIZE = 365;
+    uint256 private constant BASIS_PRECISION = 10000;
+
+    // A cyclic buffer structure for storing running total (cumulative sum)
+    // values and their respective timestamps.
+    // latestIndex points to the previously inserted value.
+    struct RunningTotalsBuffer {
+        uint256[MAX_BUFFER_SIZE] runningTotals;
+        uint256[MAX_BUFFER_SIZE] timestamps;
+        uint16 latestIndex;
+        uint256 lastValue;
```

Comment:
LRS: The funny thing is we still can't delete `lastValue`. The way we calculate average apy is by using running totals which require current apy and its time weight which we can't have yet in case of left Riemann sum. So either we save current apy and wait for next update's timestamp to get time weight or perhaps we could go with right Riemann sum.

Reply:
Sorry, my bad in mistaking how this works. Let's delete lastValue and go with a right Riemann sum.

Reply:
Starting to work on that right away.